### PR TITLE
helm: support endpoint_suffix for azure storage

### PR DESCRIPTION
**Pull request by SingleCorner**
**What this PR does / why we need it**:
According the [official document](https://grafana.com/docs/loki/latest/storage/#using-account-name-and-key), support storage_config.azure.endpoint_suffix, example blob.core.chinacloudapi.cn

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [X] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Comment by MichelHollands** [comment]
@SingleCorner Thanks for this contribution. Can you update the  production/helm/loki/Chart.yaml and update production/helm/loki/CHANGELOG.md and production/helm/loki/README.md files as per the last item in the checklist. Once that is done I'll merge this.

**Comment by SingleCorner** [comment]
> @SingleCorner Thanks for this contribution. Can you update the production/helm/loki/Chart.yaml and update production/helm/loki/CHANGELOG.md and production/helm/loki/README.md files as per the last item in the checklist. Once that is done I'll merge this.

sorry for the incomplete files , I've already updated.

**Comment by MichelHollands** [comment]
@SingleCorner Some one else got this release number earlier. Can you update it and rebase?

File: production/helm/loki/CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,12 @@ Entries should be ordered as follows:
 Entries should include a reference to the pull request that introduced the change.
 
 [//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
-## 5.8.6
+
+## 5.8.7
+
+- [ENHANCEMENT] Add loki.storage.azure.endpointSuffix to support azure private endpoint
```
Comment (JStickler) [inline production/helm/loki/CHANGELOG.md:18]:
```suggestion
- [ENHANCEMENT] Add loki.storage.azure.endpointSuffix to support Azure private endpoint
```

**Review by JStickler — APPROVED** [review]
[Docs team] LGTM.

**Pull request merged**